src/dsp_tools/fast_xmlupload/upload_files.py

```diff
@@ -324,6 +324,8 @@ def _upload_files_in_parallel(
     result: list[tuple[Path, bool]] = []
     for uploaded in as_completed(upload_jobs):
         result.append(uploaded.result())
+        if len(result) % 1000 == 0:
+            print(f"{datetime.now()}: Uploaded {len(result)} files so far, continue...")
     return result
```

Comment:
this really looks like a case for tqdm

Reply:
Don't think this would work. `as_completed(upload_jobs)` yields results from multiprocessing, i.e. waiting until more results come.

Reply:
I'm sure this could be made to work, potentially one would have to provide it the length of all expected results, if that is not a fixed-length iterable but an iterator. But admittedly, this is a different story, so no need to speculate about it now :)